telegrambot/src/main/java/ru/teamtwo/telegrambot/service/TelegramBotUpdateHandler.java

```diff
@@ -48,66 +47,91 @@ else if (userStateHandler.get(user).getState()==UserState.State.WAITING_FOR_SEAR
                 userStateHandler.get(user).setState(UserState.State.WAITING_FOR_SEARCH_START);
             }
             else if (userStateHandler.get(user).getState()==UserState.State.WAITING_FOR_SEARCH_QUERY) {
-                List<ProductDTO> queryResult = queryResultHandler.getSearchResult(message, user);
-                if (queryResult.isEmpty()) {
+                List<String> queryResult = queryResultHandler.getSearchResult(message, user);
```

Comment:
Probably better to extract 
var userState = userStateHandler.get(user).getState();

Reply:
done